frontend/src/components/CheckBox.tsx

```diff
@@ -36,11 +36,14 @@ const CheckBox = ({ label, tooltip, value, onChange, name }: Props) => {
       form: useForm(),
       /** value of checked for FormData */
       value: checkedValue,
-      /** initialize state */
-      checked: value,
-      /** when state changes */
-      onCheckedChange: (details) => onChange?.(!!details.checked),
     }),
+    {
```

Comment:
Several components have a change like this, where I had to move component props from the first argument to this second "context" argument. The difference is, the "context" is declarative and will update when changed, whereas the first argument only specifies the initual values (then the zag state machine handles the rest).